test/unit/spread.spec.ts

```diff
@@ -128,3 +129,303 @@ describe("in object literal", () => {
         `.expectToMatchJsResult();
     });
 });
+
+describe("vararg spread optimization", () => {
+    test("basic use", () => {
+        util.testFunction`
+            function pick(...args: string[]) { return args[1]; }
+            function test(...args: string[]) {
+                return pick(...args);
+            }
+            return test("a", "b", "c");
+        `
+            .expectLuaToMatchSnapshot()
```

Comment:
Not really sure, if we need expectLuaToMatchSnapshot and expectToMatchJsResult. I think if the behaviour of JS and Lua matches, it does not matter what Lua output looks like.

Reply:
You could argue in this case it does matter, since the current unoptimized lua also works, but is not exactly what we want. So we should either expect it to match a snapshot (I'm fine with this), or somehow assert the resulting code indeed has the optimizations.

Reply:
Good Point. I think checking if the generated code does not contain `unpack` could be more explicit. But I am also fine with using a snapshot.

Reply:
Yeah, this was my thinking - we need to test that the optimization occurs and that it functions correctly. Testing for unpack feels a bit messy. If something breaks it will be clearly visible in the snapshot.